src/System.ComponentModel.Composition/src/Microsoft/Internal/ReflectionServices.cs

```diff
@@ -79,9 +85,12 @@ public static string GetDisplayName(this MemberInfo member)
         internal static bool TryGetGenericInterfaceType(Type instanceType, Type targetOpenInterfaceType, out Type targetClosedInterfaceType)
         {
             // The interface must be open
-            Assumes.IsTrue(targetOpenInterfaceType.IsInterface);
-            Assumes.IsTrue(targetOpenInterfaceType.IsGenericTypeDefinition);
-            Assumes.IsTrue(!instanceType.IsGenericTypeDefinition);
+            if(!targetOpenInterfaceType.IsInterface ||
+                !targetOpenInterfaceType.IsGenericTypeDefinition ||
+                instanceType.IsGenericTypeDefinition)
+            {
+                throw new Exception();
```

Comment:
please pass the internal dignostics message to the exception here: SR.Diagnostic_InternalExceptionMessage